source/material_model/steinberger.cc

```diff
@@ -697,21 +407,6 @@ namespace aspect
                                 "prescribed " + Utilities::int_to_string(material_file_names.size()) + " material data files, but there are " +
                                 Utilities::int_to_string(this->n_compositional_fields()) + " compositional fields."));
 
-        // The Steinberger material model currently assumes that all the
-        // compositional fields correspond to materials with lookup tables.
-        // Therefore the first composition index is hard-coded as zero.
-        // If more compositional fields are needed, this parameter allows
-        // the user to declare the first compositional index which
-        // corresponds to a lookup.
-        first_composition_index = 0;
```

Comment:
So it looks like this variable wasn't used before (and also isn't used in the grain size model). In this case, I think it makes sense to just remove it rather than to keep all the lines of code in the material model to make this a potential option in the future (but still hardcode it to zero). Ideally, we probably want to implement this in the same way it is done in some other models, that it's possible to provide the names of phases that are mass/volume fractions and should be used for material averaging, rather than just providing one index.

Reply:
Yes, I agree. I was chatting with @MFraters a while ago about introducing a new labelling option for compositional fields, as the logic system we have at the moment is not very extensible.